homeassistant/components/modbus/light.py

```diff
@@ -0,0 +1,182 @@
+"""Support for Modbus lights."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from homeassistant.components.light import LightEntity
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_COMMAND_OFF,
+    CONF_COMMAND_ON,
+    CONF_LIGHTS,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+    CONF_SLAVE,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CALL_TYPE_COIL,
+    CALL_TYPE_DISCRETE,
+    CALL_TYPE_REGISTER_HOLDING,
+    CALL_TYPE_REGISTER_INPUT,
+    CONF_INPUT_TYPE,
+    CONF_STATE_OFF,
+    CONF_STATE_ON,
+    CONF_VERIFY,
+    CONF_WRITE_TYPE,
+    MODBUS_DOMAIN,
+)
+from .modbus import ModbusHub
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant, config: ConfigType, async_add_entities, discovery_info=None
+):
+    """Read configuration and create Modbus lights."""
+    lights = []
+
+    for entry in discovery_info[CONF_LIGHTS]:
+        hub: ModbusHub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        lights.append(ModbusLight(hub, entry))
+    async_add_entities(lights)
+
+
+class ModbusLight(LightEntity, RestoreEntity):
+    """Base class representing a Modbus light."""
+
+    def __init__(self, hub: ModbusHub, config: dict):
```

Comment:
hass as a parameter, so self.hass is not None. it is used in async_track_time_interval.